src/test/java/mate/academy/dao/UserDaoTest.java

```diff
@@ -0,0 +1,90 @@
+package mate.academy.dao;
+
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.dao.impl.UserDaoImpl;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.util.Optional;
+import java.util.Set;
+
+class UserDaoTest extends AbstractTest {
+    private UserDao userDao;
+    private static RoleDao roleDao;
+    private static Role roleAdmin;
+    private static Role roleUser;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[] {User.class, Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        userDao = new UserDaoImpl(getSessionFactory());
+        roleDao = new RoleDaoImpl(getSessionFactory());
+        roleAdmin = new Role(Role.RoleName.ADMIN);
+        roleUser = new Role(Role.RoleName.USER);
+        roleDao.save(roleAdmin);
+        roleDao.save(roleUser);
+    }
+
+    @Test
+    void save_ok() {
+        User expectedUser = createUser();
+        User actualUser = userDao.save(expectedUser);
+        Assertions.assertNotNull(actualUser);
+        Assertions.assertEquals(expectedUser.getId(), actualUser.getId());
+        Assertions.assertEquals(expectedUser, actualUser);
+    }
+
+    @Test
+    void findByEmailExisting_ok() {
+        User expectedUser = createUser();
+        userDao.save(expectedUser);
+        Optional<User> actualUserOptional = userDao.findByEmail(expectedUser.getEmail());
+        Assertions.assertTrue(actualUserOptional.isPresent());
+        User actualUser = actualUserOptional.get();
+        Assertions.assertEquals(expectedUser.getId(), actualUser.getId());
+        Assertions.assertEquals(expectedUser, actualUser);
+    }
+
+    @Test
+    void findByEmailNotExisting_emptyOptional() {
+        Optional<User> actualUserOptional = userDao.findByEmail("unknown@i.ua");
+        Assertions.assertTrue(actualUserOptional.isEmpty());
+    }
+
+    @Test
+    void findById_ok() {
+        User expectedUser = createUser();
+        userDao.save(expectedUser);
+        Optional<User> actualUserOptional = userDao.findById(expectedUser.getId());
+        Assertions.assertTrue(actualUserOptional.isPresent());
+        User actualUser = actualUserOptional.get();
+        Assertions.assertEquals(expectedUser.getId(), actualUser.getId());
+        Assertions.assertEquals(expectedUser.getEmail(), actualUser.getEmail());
+        Assertions.assertEquals(expectedUser.getPassword(), actualUser.getPassword());
+        //findById returns User without Roles
+    }
+
+    @Test
+    void findByIdNotExisting_emptyOptional() {
```

Comment:
```suggestion
    void findById_NotExisting_emptyOptional() {
```
Fix test method naming everywhere

There are a lot of ways to name your test methods. The main point is that they should have informative 
names and be consistent along with other developers in your team. For this task use such convention:
`<methodUnderTest>_<state>_<expectedBehavior>`; For example, if we are testing the method `register` with a `null`  user's age the test method name should be `register_nullAge_notOk`. notOk is because
the test expects the register method to throw an exception.